src/server/game/Entities/Creature/TemporarySummon.cpp

```diff
@@ -16,13 +16,39 @@ TempSummon::TempSummon(SummonPropertiesEntry const* properties, ObjectGuid owner
     Creature(isWorldObject), m_Properties(properties), m_type(TEMPSUMMON_MANUAL_DESPAWN),
     m_timer(0), m_lifetime(0)
 {
-    m_summonerGUID = owner;
+    if (owner)
```

Comment:
Hmm, do we support other objects than unit to be a summoner like TC?

Reply:
yes